lib/grandpa/votes_tracker.go

```diff
@@ -16,6 +17,7 @@ import (
 // its maximum capacity is reached.
 // It is NOT THREAD SAFE to use.
 type votesTracker struct {
+	sync.Mutex
```

Comment:
please don't embed the mutex to avoid exposing `.Lock()` and `.Unlock()` methods (even though it's not exported for now, it might be in the future). It also makes asserting the entire object cumbersome in tests as you've seen.

Reply:
not embedding does not fix this. I still can't use `assert.Equal(t, expected, vt)` as long as there is mutex in the struct